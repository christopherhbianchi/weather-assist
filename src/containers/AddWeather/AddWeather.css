.add-weather {
  background: #4286f4;
  color: white;
  border-radius: 25px;
  padding-top: 5px;
  padding-bottom: 15px;
}
