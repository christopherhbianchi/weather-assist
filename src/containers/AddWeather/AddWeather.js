import React, { Component } from 'react';
import { connect } from 'react-redux';
import './AddWeather.css';
import PropTypes from 'prop-types';
//import components below
import { createWeather } from '../../actions/weatherActions.js';
const CityCoordinates = require('../../resources/CityCoordinates.js');

export class AddWeather extends Component {
  constructor(){
    super();
    this.state = {
      locationToAdd: CityCoordinates.LasVegas
    };
  }//closes constructor

  handleChange = async (event) => {
    console.log(CityCoordinates);
    console.log('AddWeather.js handleChange: ' + event.target.value);
    var location = 0;
    for (const prop1 in CityCoordinates){
      if (event.target.value === prop1){
        location = CityCoordinates[prop1]; //location would have "NewYork" object or something
      }
    }
    console.log('location after for in: ' + location);
    console.log('location.latitude after for in: ' + location.latitude);
    await this.setState({ locationToAdd: location },
    () => { console.log('setState callback in handleChange: ' + this.state.locationToAdd.latitude) });//closes setState

  }//closes handleChange

  handleSubmit = (event) => {
    event.preventDefault();
    var locToAdd = this.state.locationToAdd.name; //name

    const weatherArr = this.props.weather;//if the key exists already, do nothing
    const found = this.props.weather.filter(city => city.name === locToAdd);
    if (!found.length) this.props.createWeather(this.state);
  }//handleSubmit

  render(){
    return (
      <div className='add-weather'>
        <h3 className='bannerAddWeather'>Select City</h3>
        <form className='addCityForm'>
          <select className='select' onChange={this.handleChange}>
            <option value='LasVegas'>Las Vegas</option>
            <option value='NewYork'>New York</option>
            <option value='Miami'>Miami</option>
            <option value='LosAngeles'>Los Angeles</option>
          </select>
          <button className='addCityButton' onClick={this.handleSubmit}>Add this city</button>
        </form>
      </div>
    );//closes return
  }//closes render
}//closes class


export const mapStateToProps = (state) => ({
  weather: state.weather
});
export const mapDispatchToProps = (dispatch) => ({
  createWeather: cityCoordinates => dispatch(createWeather(cityCoordinates))
});

AddWeather.propTypes = {
  weather: PropTypes.array,
  createWeather: PropTypes.func
};

export default connect(mapStateToProps, mapDispatchToProps)(AddWeather);
