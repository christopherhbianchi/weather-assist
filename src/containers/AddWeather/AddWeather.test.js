import React from 'react';
import { shallow } from 'enzyme';
import { AddWeather, mapStateToProps, mapDispatchToProps } from './AddWeather.js';
const CityCoordinates = require('../../resources/CityCoordinates.js');

describe('<AddWeather />', () => {

  const renderedComponent = shallow(
    <AddWeather
      locationToAdd={ {latitude: 1, longitude: 1} }/>
  );//shallow


  it('should match snapshot', () => {
    expect(renderedComponent).toMatchSnapshot();
  });//it

  it('should have a default state', () => {
    const expected = { locationToAdd: CityCoordinates.LasVegas };
    expect(renderedComponent.state()).toEqual(expected);
  });//it

  it('should map the store correctly', () => {
    const mockStore = { weather: {} };
    const mapped = mapStateToProps(mockStore);

    expect(mapped).toEqual(mockStore);
  });//it

  it('should dispatch createWeather', () => {
    const mockDispatch = jest.fn();
    const mapped = mapDispatchToProps(mockDispatch);

    //this method takes a locationToAdd object, which then has latitude and longitude properties
    mapped.createWeather({ locationToAdd: {latitude: 1, longitude: 1}});

    expect(mockDispatch).toHaveBeenCalled();
  });//it







});//describe
