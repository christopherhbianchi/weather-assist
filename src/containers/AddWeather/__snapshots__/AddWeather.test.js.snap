// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddWeather /> should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AddWeather
    locationToAdd={
        Object {
            "latitude": 1,
            "longitude": 1,
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3
          className="bannerAddWeather"
>
          Select City
</h3>,
        <form
          className="addCityForm"
>
          <select
                    className="select"
                    onChange={[Function]}
          >
                    <option
                              value="LasVegas"
                    >
                              Las Vegas
                    </option>
                    <option
                              value="NewYork"
                    >
                              New York
                    </option>
                    <option
                              value="Miami"
                    >
                              Miami
                    </option>
                    <option
                              value="LosAngeles"
                    >
                              Los Angeles
                    </option>
          </select>
          <button
                    className="addCityButton"
                    onClick={[Function]}
          >
                    Add this city
          </button>
</form>,
      ],
      "className": "add-weather",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Select City",
          "className": "bannerAddWeather",
        },
        "ref": null,
        "rendered": "Select City",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <select
              className="select"
              onChange={[Function]}
>
              <option
                            value="LasVegas"
              >
                            Las Vegas
              </option>
              <option
                            value="NewYork"
              >
                            New York
              </option>
              <option
                            value="Miami"
              >
                            Miami
              </option>
              <option
                            value="LosAngeles"
              >
                            Los Angeles
              </option>
</select>,
            <button
              className="addCityButton"
              onClick={[Function]}
>
              Add this city
</button>,
          ],
          "className": "addCityForm",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <option
                  value="LasVegas"
>
                  Las Vegas
</option>,
                <option
                  value="NewYork"
>
                  New York
</option>,
                <option
                  value="Miami"
>
                  Miami
</option>,
                <option
                  value="LosAngeles"
>
                  Los Angeles
</option>,
              ],
              "className": "select",
              "onChange": [Function],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Las Vegas",
                  "value": "LasVegas",
                },
                "ref": null,
                "rendered": "Las Vegas",
                "type": "option",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "New York",
                  "value": "NewYork",
                },
                "ref": null,
                "rendered": "New York",
                "type": "option",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Miami",
                  "value": "Miami",
                },
                "ref": null,
                "rendered": "Miami",
                "type": "option",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Los Angeles",
                  "value": "LosAngeles",
                },
                "ref": null,
                "rendered": "Los Angeles",
                "type": "option",
              },
            ],
            "type": "select",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Add this city",
              "className": "addCityButton",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Add this city",
            "type": "button",
          },
        ],
        "type": "form",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3
            className="bannerAddWeather"
>
            Select City
</h3>,
          <form
            className="addCityForm"
>
            <select
                        className="select"
                        onChange={[Function]}
            >
                        <option
                                    value="LasVegas"
                        >
                                    Las Vegas
                        </option>
                        <option
                                    value="NewYork"
                        >
                                    New York
                        </option>
                        <option
                                    value="Miami"
                        >
                                    Miami
                        </option>
                        <option
                                    value="LosAngeles"
                        >
                                    Los Angeles
                        </option>
            </select>
            <button
                        className="addCityButton"
                        onClick={[Function]}
            >
                        Add this city
            </button>
</form>,
        ],
        "className": "add-weather",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Select City",
            "className": "bannerAddWeather",
          },
          "ref": null,
          "rendered": "Select City",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <select
                className="select"
                onChange={[Function]}
>
                <option
                                value="LasVegas"
                >
                                Las Vegas
                </option>
                <option
                                value="NewYork"
                >
                                New York
                </option>
                <option
                                value="Miami"
                >
                                Miami
                </option>
                <option
                                value="LosAngeles"
                >
                                Los Angeles
                </option>
</select>,
              <button
                className="addCityButton"
                onClick={[Function]}
>
                Add this city
</button>,
            ],
            "className": "addCityForm",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <option
                    value="LasVegas"
>
                    Las Vegas
</option>,
                  <option
                    value="NewYork"
>
                    New York
</option>,
                  <option
                    value="Miami"
>
                    Miami
</option>,
                  <option
                    value="LosAngeles"
>
                    Los Angeles
</option>,
                ],
                "className": "select",
                "onChange": [Function],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Las Vegas",
                    "value": "LasVegas",
                  },
                  "ref": null,
                  "rendered": "Las Vegas",
                  "type": "option",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "New York",
                    "value": "NewYork",
                  },
                  "ref": null,
                  "rendered": "New York",
                  "type": "option",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Miami",
                    "value": "Miami",
                  },
                  "ref": null,
                  "rendered": "Miami",
                  "type": "option",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Los Angeles",
                    "value": "LosAngeles",
                  },
                  "ref": null,
                  "rendered": "Los Angeles",
                  "type": "option",
                },
              ],
              "type": "select",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Add this city",
                "className": "addCityButton",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Add this city",
              "type": "button",
            },
          ],
          "type": "form",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
